shell/shared/grain.js

```diff
@@ -1300,6 +1304,7 @@ Router.map(function () {
     },
 
     onStop: function () {
+      this.state.set("beforeActionHookRan", undefined);
```

Comment:
Yeah, I don't ​know if I have a strong opinion either way. Consider me +0
on this commit as-is!